NitroxClient/MonoBehaviours/ThrottledBuilder.cs

```diff
@@ -274,6 +278,13 @@ private static GameObject FindFinishedPiece(Transform cellTransform)
             return null;
         }
 
+        private static bool AreSameTechType(TechType actual, TechType expected)
+        {
+            return actual.Equals(expected) || // normal case
+                   (actual.Equals(TechType.BaseConnector) && expected.Equals(TechType.BaseCorridor)) || // a specific case
+                   expected.ToString().Contains(actual.ToString()); // for cases like: actual = BaseCorridor but expected = BaseCorridorT
```

Comment:
Maybe you should split into multiple line instead of fitting logic  inside a single return, it's quite hard to read right now